torch/_inductor/codegen/triton.py

```diff
@@ -782,10 +783,15 @@ def indexing(
         if self._load_mask:
             mask_vars.add(self._load_mask)
 
-        if mask_vars == {"xmask"} and index == 0 and self.range_trees[0].numel == 1:
-            # This causes a triton error:
-            # https://github.com/openai/triton/issues/633
-            mask_vars = set()
+        for tree in self.range_trees:
+            # Masks are superfluous if we only have one element
+            # or if numel is a multiple of BLOCK
+            # (We use the fact that BLOCK is required by triton to be a power of 2)
+            if tree.prefix.upper() not in config.triton.max_block:
+                continue
+            max_block = config.triton.max_block[tree.prefix.upper()]
+            if tree.numel == 1 or tree.numel % max_block == 0:
```

Comment:
cc @chillee, this modulo will probably never be true with symbolic shapes? To enable this mask clearing we'd need to look at size hints and set up corresponding guards?

Reply:
Yeah, that sounds right.